Week-6/mapSetEcommerce/script.js

```diff
@@ -0,0 +1,38 @@
+const productViews = new WeakMap();
+const cartItems = new WeakSet();
+
+const incrementProductViews = (productId) => {
+  if(productViews.has(productId)) {
+    productViews.set(productId, productViews.get(productId) + 1);
+  } else {
+    productViews.set(productId, 1);
+  }
+  console.log(productViews.get(productId))
+};
+
+const addToCart = (productId) => {
+  if(cartItems.has(productId)) {
+    console.log("Product already in cart");
+  } else {
+    cartItems.add(productId);
+    console.log("product added to cart.");
+  }
+}
+const earphone = {
+  brand: "sony",
+  model: "wf-700n"
+}
+
+const phone = {
+  brand: "xiaomi",
+  model: "11T pro"
+}
+incrementProductViews(earphone)
+incrementProductViews(earphone)
+incrementProductViews(phone)
+incrementProductViews(phone)
+
+addToCart(earphone)
```

Comment:
1. The idea to use WeakMap for productViews may not be appropriate because WeakMap keys are usually DOM objects, and you're using plain objects here. A regular Map might be a better fit unless you have a specific reason for weak references.
2. The addToCart function currently prints to the console. It could also return a status or trigger some other action to give feedback in a real-world scenario.